### PR TITLE
pre-commit: integrate with cndp

### PR DESCRIPTION
**Pull request by maryamtahhan**
Install pre-commit for CNDP + its configuration.
Remove old githook.

### CODE REVIEW

File: doc/api/doxy-api-index.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-API {#index}
-===
+# API {#index}
```
Comment (maryamtahhan) [inline doc/api/doxy-api-index.md:1]: Restore this file and add it to the ignore list

File: examples/dlb_test/README.md [hunk 2]
```diff
@@ -17,15 +17,16 @@ The tarball contains a DLB driver and libdlb (user space library).
...
-Create a directory named dlb and copy all the header files from 
-$BASE_DLB_DIR/libdlb*.h into /usr/local/include/dlb
-CNDP will then link the library to the app if added as a dependency in the 
-example's meson file
-  deps += [libdlb]
+
+Create a directory named dlb and copy all the header files from
+$BASE_DLB_DIR/libdlb\*.h into /usr/local/include/dlb CNDP will then link the
+library to the app if added as a dependency in the example's meson file deps +=
+\[libdlb\]
```
Comment (maryamtahhan) [inline examples/dlb_test/README.md:32]: Double check this

File: lang/rs/wireguard/patch/0002-Remove-LD_PRELOAD-from-cndprustwg-script.patch [hunk 3]
```diff
@@ -18,9 +18,9 @@ index f390b63..d23f822 100644
 --- a/src/platform/linux/cndp/README.md
```
Comment (maryamtahhan) [inline lang/rs/wireguard/patch/0002-Remove-LD_PRELOAD-from-cndprustwg-script.patch:18]: Restore patch files

File: lang/rs/wireguard/patch/0003-Update-latest-crypto-library-version.patch [hunk 4]
```diff
@@ -32,15 +32,15 @@ index 2941071..51cd887 100644
  dependencies = [
```
Comment (maryamtahhan) [inline lang/rs/wireguard/patch/0003-Update-latest-crypto-library-version.patch:32]: restore

File: test/common/test.h [hunk 5]
```diff
@@ -182,9 +182,6 @@ void add_test_command(struct test_command *t);
         .command  = CNE_STR(cmd),                    \
         .callback = func,                            \
     };                                               \
-    CNE_INIT(test_register_##cmd)                    \
-    {                                                \
-        add_test_command(&test_struct_##cmd);        \
-    }
+    CNE_INIT(test_register_##cmd) { add_test_command(&test_struct_##cmd); }
```
Comment (maryamtahhan) [inline test/common/test.h:185]: Double check clang formatting

File: usrtools/txgen/README.md [hunk 6]
```diff
@@ -1,21 +1,22 @@
 # TXGen - Traffic Generator powered by CNDP
 
-=====================================================
+\=====================================================
```
Comment (maryamtahhan) [inline usrtools/txgen/README.md:3]:
```suggestion
---
```

**Comment by maryamtahhan** [comment]
Sorry for the noise - it took a while to find the right recipe of clang versions between pre-commit hooks and the github action

**Comment by maryamtahhan** [comment]
:) Markdown linter disconnect now :(

**Review by KeithWiles — APPROVED** [review]
Thanks Maryam, that is a lot of work and it looks good to me.

**Comment by KeithWiles** [comment]
Should this get merged now or will you be fixing the Lint Code Base issues?

**Comment by maryamtahhan** [comment]
I will fix them :)

**Comment by maryamtahhan** [comment]
Linting issues are now fixed. I just rebased there and kicked off the github actions again. As soon as they complete we should be good. 

Sorry again for all the noise :( I couldn't reproduce the issues locally

**Comment by KeithWiles** [comment]
No problem it is a lot of work, Thanks

**Pull request closed without merging**